--- libdbus-1-qt3-0.8.1/configure.in	2007-12-17 11:34:08.000000000 +0000
+++ libdbus-1-qt3-0.8.1/configure.in	2008-01-26 15:48:40.000000000 +0000
@@ -27,10 +27,10 @@
   KDE_PKG_CHECK_MODULES( DBUS, "dbus-1", [ DBUS_INCS=$DBUS_CFLAGS; dbus_inc=FOUND; dbus_lib=FOUND; ] , AC_MSG_RESULT( Nothing found on PKG_CONFIG_PATH ) )
 
   if test $dbus_inc != FOUND || test $dbus_lib != FOUND ; then
-    search_incs="/usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+    search_incs="/usr/include /usr/include/dbus-1.0"
     AC_FIND_FILE(dbus/dbus.h, $search_incs, dbus_incdir)
 
-    search_incs_arch_deps="/usr/lib/dbus-1.0/include /usr/local/lib/dbus-1.0/include"
+    search_incs_arch_deps="${libdir}/dbus-1.0/include"
     AC_FIND_FILE(dbus/dbus-arch-deps.h, $search_incs_arch_deps, dbus_incdir_arch_deps)
 
     if [test -r $dbus_incdir/dbus/dbus.h] && [test -r $dbus_incdir_arch_deps/dbus/dbus-arch-deps.h] ; then
@@ -38,7 +38,7 @@
       dbus_inc=FOUND
     fi
 
-    search_libs="/lib /usr/lib /usr/local/lib"
+    search_libs="${libdir}"
     AC_FIND_FILE(libdbus-1.so, $search_libs, dbus_libdir)
 
     if test -r $dbus_libdir/libdbus-1.so ; then
