diff -uNr dbus-1-qt3-0.2/dbus/qdbusconnection.h dbus-1-qt3-0.2.patched/dbus/qdbusconnection.h
--- dbus-1-qt3-0.2/dbus/qdbusconnection.h	2006-01-22 22:57:15.000000000 +0000
+++ dbus-1-qt3-0.2.patched/dbus/qdbusconnection.h	2006-08-18 13:48:23.730760750 +0000
@@ -118,6 +118,7 @@
  */
 
 #include "dbus/qdbusmacros.h"
+#include <dbus/dbus.h>
 #include <qstring.h>
 
 class QDBusConnectionPrivate;
@@ -607,6 +608,12 @@
      */
     QT_STATIC_CONST char *default_connection_name;
 
+    /**
+     * That enables integration of a given dbus connection in the qt main loop.
+     * It is needed to make a Qt application that recieves HAL callbacks.
+     */
+    void dbus_connection_setup_with_qt_main(DBusConnection *connection);
+
 private:
     QDBusConnectionPrivate *d;
 };
diff -uNr dbus-1-qt3-0.2/qdbusconnection.cpp dbus-1-qt3-0.2.patched/qdbusconnection.cpp
--- dbus-1-qt3-0.2/qdbusconnection.cpp	2006-01-22 22:57:16.000000000 +0000
+++ dbus-1-qt3-0.2.patched/qdbusconnection.cpp	2006-08-18 13:48:31.555249750 +0000
@@ -382,4 +382,9 @@
     return !d->handleError();
 }
 
+void QDBusConnection::dbus_connection_setup_with_qt_main(DBusConnection *connection)
+{
+    d->setConnection(connection);
+}
+
 #include "qdbusconnection.moc"
